### PR TITLE
refactor: Implement better abstraction for use in Secret Provider

### PR DESCRIPTION
**Pull request by lenny-intel**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [X] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
original implementation of abstraction did follow how other module's abstract with implemented.

Issue Number: #69

## What is the new behavior?
Abstraction is more consistent with other modules abstractions, i.e Registry, Messaging, etc
Also created proper `Mockery` mocks of interfaces so consumers can mock these interfaces in unit tests.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Are there any new imports or modules? If so, what are they used for and why?
no

## Other information

### CODE REVIEW

**Review by bnevis-i — APPROVED** [review]
LGTM

File: pkg/interfaces.go [hunk 1]
```diff
@@ -22,19 +22,4 @@ type Caller interface {
 }
 
 // tokenExpiredCallback is the callback function to handle the case when the vault token has already expired
```
Comment (jim-wang-intel) [inline pkg/interfaces.go:24]: comment is out of dated, update

File: pkg/types/config.go [hunk 2]
```diff
@@ -13,7 +13,7 @@
  *******************************************************************************/
 
 // Package vault defines structs that will be used frequently by clients which utilize HTTP transport.
```
Comment (jim-wang-intel) [inline pkg/types/config.go:15]: update the out-of-date package comment

**Review by jim-wang-intel — COMMENTED** [review]
code-wise, LGTM; just a couple of comments out of date.

File: pkg/interfaces.go (hunk 1 above)
Comment (lenny-intel) [reply pkg/interfaces.go:24]: You  mean `tokenExpiredCallback ` => `TokenExpiredCallback` ?

File: pkg/types/config.go (hunk 2 above)
Comment (lenny-intel) [reply pkg/types/config.go:15]: Remove package comment. Never thought they are usefull.

File: pkg/interfaces.go (hunk 1 above)
Comment (jim-wang-intel) [reply pkg/interfaces.go]: yea

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**